src/main/php/lang/ast/emit/PHP.class.php

```diff
@@ -661,18 +668,32 @@ protected function emitNew($result, $new) {
   }
 
   protected function emitNewClass($result, $new) {
+    array_unshift($result->meta, []);
+
     $result->out->write('new class(');
     $this->emitArguments($result, $new->arguments);
     $result->out->write(')');
 
     $new->definition->parent && $result->out->write(' extends '.$new->definition->parent);
     $new->definition->implements && $result->out->write(' implements '.implode(', ', $new->definition->implements));
     $result->out->write('{');
+
+    // Initialize meta data in constructor
+    if (isset($new->definition->body['__construct()'])) {
+      array_unshift($new->definition->body['__construct()']->body, new Code('self::__init()'));
+    } else {
+      $new->definition->body['__construct()']= new Method([], '__construct', new Signature([], null), [
```

Comment:
This will break if a parent constructor exists, we would need to check this, or write something like:

```php
public function __construct(... $args) {
  if (method_exists(parent::class, '__construct')) {
   parent::__construct(...$args);
  }
  self::__init();
}
```